client/components/domains/registrant-extra-info/ca-form.jsx

```diff
@@ -0,0 +1,107 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import { noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getContactDetailsCache } from 'state/selectors';
+import { updateContactDetailsCache } from 'state/domains/management/actions';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormSelect from 'components/forms/form-select';
+import FormCheckbox from 'components/forms/form-checkbox';
+
+const defaultValues = {
+	legalType: 'CCW',
+	isVisible: true,
+	agreementAccepted: false,
+	onSubmit: noop,
+};
+
+class RegistrantExtraInfoCaForm extends React.PureComponent {
+	static propTypes = {
+		agreementAccepted: PropTypes.bool,
+		legalType: PropTypes.string,
+		isVisible: PropTypes.bool,
+		onSubmit: PropTypes.func,
+	}
+
+	constructor( props ) {
+		super( props );
+
+		this.props.contactDetails.extra = {
+			...defaultValues,
+			...this.props.contactDetails.extra
+		};
+	}
+
+	handleChangeEvent = ( event ) => {
+		this.props.updateContactDetailsCache( {
+			extra: { [ event.target.id ]: event.target.value },
+		} );
+	}
+
+	render() {
+		const {
+			legalType,
+			translate,
+			ciraAccepted
+		} = { ...defaultValues, ...this.props };
+		const agreementUrl = 'https://services.cira.ca/agree/agreement/agreementVersion2.0.jsp';
+
+		return (
+			<form className="registrant-extra-info__form">
+				<h1 className="registrant-extra-info__form-title">
+					{ translate(
+						'Registering a .CA domain'
+					) }
+				</h1>
+				<p className="registrant-extra-info__form-desciption">
+					{ translate(
+						'Almost done! We need some extra details to register domains ending in ".ca".'
```

Comment:
We should combine these two strings to one string with a placeholder for (.fr|.ca|...)